src/Airship/Internal/Decision.hs

```diff
@@ -131,6 +131,32 @@ writeCacheTags Resource{..} = lift $ do
        Nothing -> return ()
        Just d  -> addResponseHeader ("Last-Modified", utcTimeToRfc1123 d)
 
+-- | Immediately halts processing with the provided 'Status' code. The response body is
+-- | optionally set based on the error responses provided by the application and resource.
+-- | The contents of the 'Webmachine''s response body will be streamed back to the client.
+-- | This is a shortcut for constructing a 'Response' with 'getResponseHeaders' and 'getResponseBody'
+-- | and passing that response to 'finishWith'.
+haltWithErrorResponse :: Monad m => HTTP.Status -> Resource m -> Webmachine m a
+haltWithErrorResponse status Resource{..} = do
+    req <- request
+    let reqHeaders = requestHeaders req
+        acceptStr = lookup HTTP.hAccept reqHeaders
+        errBodies = map dupContentType <$> M.lookup HTTP.status404 errorResponses
+        mResp = join $ mapAcceptMedia <$> errBodies <*> acceptStr
+    when (isJust mResp) $ do
+       let (Just (ct, body)) = mResp
```

Comment:
Would suggest pattern matching or `forM_` here rather than the `isJust` and unsafe  pattern matching